common/queue_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewQueue(t *testing.T) {
+	q := NewQueue[int]()
+
+	assert.NotNil(t, q, "NewQueue() should return a non-nil queue.")
+	assert.True(t, q.IsEmpty(), "A new queue should be empty.")
+	assert.Equal(t, 0, q.Len(), "A new queue should have a size of 0.")
+}
+
+func TestQueue_Push(t *testing.T) {
+	q := NewQueue[int]()
+
+	q.Push(4)
+	q.Push(5)
+
+	assert.Equal(t, 4, q.Peek())
+	assert.False(t, q.IsEmpty())
+}
+
+func TestQueue_Pop(t *testing.T) {
+	q := NewQueue[int]()
+	q.Push(4)
+	q.Push(5)
+	require.Equal(t, 4, q.Peek())
+	require.False(t, q.IsEmpty())
+
+	val := q.Pop()
+	assert.Equal(t, 4, val)
+	assert.Equal(t, 5, q.Peek())
+
+	val = q.Pop()
+	assert.Equal(t, 5, val)
+	assert.True(t, q.IsEmpty())
+}
+
+func TestQueue_PopNewQueue(t *testing.T) {
+	q := NewQueue[int]()
+	assert.True(t, q.IsEmpty(), "New queue should be empty.")
```

Comment:
The fact that NewQueue is empty should have a test of its own. Let's not have this assert here.

Reply:
Changing assert -> require.

Reply:
Please refer to these guidelines: https://screenshot.googleplex.com/A42iSeX3o8MRhGR

Reply:
Thanks, Got, it!
Not required here, as deleted this test case.